### PR TITLE
OF-1296 GroupManager.getSharedGroups returning old values

### PR DESCRIPTION
**Pull request by aberenguel**

### CODE REVIEW

File: src/java/org/jivesoftware/openfire/group/GroupManager.java [hunk 1]
```diff
@@ -713,4 +765,15 @@ private void evictCachedPaginatedGroupNames() {
             }
         }
     }
+
+    private void evictCachedUserSharedGroups() {
+        synchronized (USER_SHARED_GROUPS_KEY.intern()) {
+            for(Map.Entry<String, Object> entry : groupMetaCache.entrySet())
+            {
+                if (entry.getKey().startsWith(USER_SHARED_GROUPS_KEY)) {
+                    groupMetaCache.remove(entry.getKey());
```
Comment (guusdk) [inline src/java/org/jivesoftware/openfire/group/GroupManager.java]: It's probably unwise to remove items from the collection while iterating over it.

**Comment by akrherz** [comment]
The current travis build failure seems to be
```
/home/travis/build/igniterealtime/Openfire/src/java/org/jivesoftware/openfire/group/GroupManager.java:[21,31] package org.apache.commons.lang does not exist
```

**Comment by GregDThomas** [comment]
I recently updated all the Apache-commons libraries, so it probably needs updating to org.apache.commons.lang3

**Comment by akrherz** [comment]
thanks @GregDThomas , I updated and rebased

**Comment by akrherz** [comment]
@guusdk we good to go now on this?

**Comment by akrherz** [comment]
rebased, will see what CI says

**Comment by akrherz** [comment]
As per weekly discussion, going to take this.  Thanks for the patience @aberenguel

**Pull request merged**